lightning/src/routing/onion_message.rs

```diff
@@ -60,6 +60,9 @@ pub fn find_path<L: Deref, GL: Deref>(
 			return Ok(reverse_path(visited, our_node_id, dest_node_id, logger)?)
 		}
 		if let Some(node_info) = network_nodes.get(&node_id) {
+			if let Some(node_ann) = &node_info.announcement_info {
+				if !node_ann.features.supports_onion_messages() { continue; }
+			} else if node_id != our_node_id { continue; }
```

Comment:
I think this line is particularly hard to understand, could you add a comment? Or even, do this check 3 lines before even accessing `announcement_info`.

Reply:
Lmk if the new version + comment make sense